crates/sdk/src/signing.rs

```diff
@@ -703,25 +702,170 @@ fn format_outputs(output: &mut Vec<String>) {
+                let mut map: HashMap<&Address, DenominatedAmount> =
+                    HashMap::new();
+
+                for transfer in &transfers.0 {
+                    match address {
+                        TransferSide::Source(source)
+                            if source == &transfer.source =>
+                        {
+                            Self::update_token_amount_map(
+                                &mut map,
+                                &transfer.token,
+                                transfer.amount,
+                            )?;
+                        }
+                        TransferSide::Target(target)
+                            if target == &transfer.target =>
+                        {
+                            Self::update_token_amount_map(
+                                &mut map,
+                                &transfer.token,
+                                transfer.amount,
+                            )?;
+                        }
+                        _ => (),
+                    }
+                }
+
+                map
+            }
+            TokenTransfer::Shielded => Default::default(),
+            TokenTransfer::Shielding(transfers) => {
+                let mut map: HashMap<&Address, DenominatedAmount> =
+                    HashMap::new();
+
+                if let TransferSide::Source(source_addr) = address {
+                    for transfer in &transfers.data {
+                        if &transfer.source == source_addr {
+                            Self::update_token_amount_map(
+                                &mut map,
+                                &transfer.token,
+                                transfer.amount,
+                            )?;
+                        }
+                    }
+                }
+
+                map
+            }
+            TokenTransfer::Unshielding(transfers) => {
+                let mut map: HashMap<&Address, DenominatedAmount> =
+                    HashMap::new();
+
+                if let TransferSide::Target(target_addr) = address {
+                    for transfer in &transfers.data {
+                        if &transfer.target == target_addr {
+                            Self::update_token_amount_map(
+                                &mut map,
+                                &transfer.token,
+                                transfer.amount,
+                            )?;
+                        }
+                    }
+                }
+
+                map
+            }
+        })
+    }
+
+    fn update_token_amount_map<'a>(
+        map: &mut HashMap<&'a Address, DenominatedAmount>,
+        token: &'a Address,
+        amount: DenominatedAmount,
+    ) -> Result<(), Error> {
+        match map.get_mut(token) {
+            Some(prev_amount) => {
+                *prev_amount =
+                    prev_amount.checked_add(amount).ok_or_else(|| {
```

Comment:
shouldn't we use the `checked!` macro here?

Reply:
Done in 280b8e245005d7cf8a15d16d650eafe380e72ab7